docs_user/modules/assembly_ospdo-scale-down-pre-database-adoption.adoc

```diff
@@ -0,0 +1,215 @@
+      name: enp7s0.50
+      state: up
+      type: vlan
+      vlan:
+        base-iface: enp7s0
+        id: 50
+        reorder-headers: true
+      ipv4:
+        address:
+        - ip: 172.20.0.7
+          prefix-length: 24
+        enabled: true
+        dhcp: false
+      ipv6:
+        enabled: false
+    - description: Configuring Bridge br-ctlplane with interface enp7s0
+      name: br-ctlplane
+      mtu: 1500
+      type: linux-bridge
+      state: up
+      bridge:
+        options:
+          stp:
+            enabled: false
+        port:
+          - name: enp1s0
+            vlan: {}
+      ipv4:
+        address:
+        - ip: 172.22.0.53
+          prefix-length: 24
+        enabled: true
+        dhcp: false
+      ipv6:
+        enabled: false
+    - bridge:
+        options:
+          stp:
+            enabled: false
+        port:
+        - name: enp6s0
+      description: Linux bridge with enp6s0 as a port
+      ipv4:
+        enabled: false
+      ipv6:
+        enabled: false
+      mtu: 1500
+      name: br-external
+      state: up
+      type: linux-bridge
+  nodeSelector:
+    kubernetes.io/hostname: <ostest-master-node>
+    node-role.kubernetes.io/worker: ""
+EOF
+
+oc apply -f /tmp/node3_nncp.yaml
+----
++
+. Get remaining OSPdO operands and delete everything, except DO NOT DELETE OpenStackBaremetalSets nor OpenStackProvisionServers yet
++
+----
+for i in $(oc get crd | grep osp-director | grep -v baremetalset | grep -v provisionserver | awk {'print $1'}); do echo Deleting $i...; oc delete $i -n openstack --all; done
+----
++
+. Scale down OSPdO to 0 replicas
++
+----
+ospdo_csv_ver=$(oc get csv -n openstack -l operators.coreos.com/osp-director-operator.openstack -o json | jq -r '.items[0].metadata.name')
+oc patch csv -n openstack $ospdo_csv_ver --type json   -p="[{"op": "replace", "path": "/spec/install/spec/deployments/0/spec/replicas", "value": "0"}]"
+----
++
+. Remove the webhooks from OSPdO
++
+----
+oc patch csv $ospdo_csv_ver -n openstack --type json -p="[{"op": "remove", "path": "/spec/webhookdefinitions"}]"
+----
++
+. Remove the finalizer from the OSPdO OpenStackBaremetalSet
++
+----
+oc patch openstackbaremetalsets.osp-director.openstack.org -n openstack compute --type json -p="[{"op": "remove", "path": "/metadata/finalizers"}]"
+----
++
+. Now delete the OpenStackBaremetalSet and OpenStackProvisionServer
++
+----
+oc delete openstackbaremetalsets.osp-director.openstack.org -n openstack --all
+oc delete openstackprovisionservers.osp-director.openstack.org -n openstack --all
+----
++
+Annotate all OSP compute BMHs so that Metal3 no longer handles them
++
+* Do this for each OSP compute BMH
++
+----
+compute_bmh_list=$(oc get bmh -n openshift-machine-api |grep compute|awk '{printf $1 " "}')
+
+for bmh_compute in $compute_bmh_list;do oc annotate bmh -n openshift-machine-api $bmh_compute baremetalhost.metal3.io/detached="";
+oc delete bmh -n openshift-machine-api $bmh_compute;done
+----
++
```

Comment:
```suggestion
. Annotate each {OpenStackShort} Compute `BareMetalHost` resource so that Metal3 does not start the node:
+
----
compute_bmh_list=$(oc get bmh -n openshift-machine-api |grep compute|awk '{printf $1 " "}')

for bmh_compute in $compute_bmh_list;do oc annotate bmh -n openshift-machine-api $bmh_compute baremetalhost.metal3.io/detached="";
oc delete bmh -n openshift-machine-api $bmh_compute;done
----

```
@pinikomarov Is my rewrite accurate?

Reply:
ok , and yes